src/cashlib/cashlib.cpp

```diff
@@ -76,24 +110,13 @@ SLAPI int Bin2Hex(unsigned char *val, int length, char *result, unsigned int res
     return 1;
 }
 
-
-/** Return random bytes from cryptographically acceptable random sources */
-SLAPI int RandomBytes(unsigned char *buf, int num)
-{
-    if (RAND_bytes(buf, num) != 1)
-    {
-        memset(buf, 0, num);
-        return 0;
-    }
-    return num;
-}
-
 /** Given a private key, return its corresponding public key */
 SLAPI int GetPubKey(unsigned char *keyData, unsigned char *result, unsigned int resultLen)
 {
     if (!sigInited)
     {
         sigInited = true;
+        SHA256AutoDetect();
```

Comment:
Seems unrelated to this PR

Reply:
This PR also fixes some issues with cashlib that were discovered while using it in Android and Java.